src/test/java/org/folio/rest/api/OaiPmhViewTest.java

```diff
@@ -385,6 +402,74 @@ public void testResponseStatus400WhenRequestingWithInvalidUntilDate()
     });
   }
 
+  @Test
+  public void test500(TestContext testContext) {
+    RoutingContext routingContext = mock(RoutingContext.class);
+    when(routingContext.response()).thenReturn(mock(HttpServerResponse.class));
+    AbstractInstanceRecordsAPIImpl.fetchRecordsByQuery("SELECT 1",
+        routingContext, null, testContext.asyncAssertSuccess(response -> {
+          assertThat(response.getStatus(), is(500));
+        }));
+  }
+
+  @Test
+  public void testTcpClose(TestContext testContext) {
+    HttpServerResponse httpServerResponse = mock(HttpServerResponse.class);
+    when(httpServerResponse.headWritten()).thenReturn(true);
+    RoutingContext routingContext = mock(RoutingContext.class);
+    when(routingContext.response()).thenReturn(httpServerResponse);
+    AbstractInstanceRecordsAPIImpl.fetchRecordsByQuery("SELECT 1",
+        routingContext, null, testContext.asyncAssertSuccess(response -> {
+          assertThat(response, is(nullValue()));
+          verify(httpServerResponse).close();
+        }));
+  }
+
+  private HttpServerResponse succeedingHttpServerResponse() {
```

Comment:
Under what circumstances should the response be closed?

From the investigation I've done, this is triggered by a null pointer exception thrown because of the null parameter. However the setup includes setting that the response headers had already been written.

If a non-null value is used for the parameter (using an empty tuple like in the other tests) the expected behaviour does not occur.

Which scenario was this test intending to describe?

Reply:
This method returns a mock. The next two unit tests use it.

Reply:
Apologies, this comment should have been on the TCP close test